Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_mesh_to_polygon_soup.h

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2020 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Mael Rouxel-Labbé
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_POLYGON_MESH_TO_POLYGON_SOUP_H
+#define CGAL_POLYGON_MESH_PROCESSING_POLYGON_MESH_TO_POLYGON_SOUP_H
+
+#include <CGAL/license/Polygon_mesh_processing/repair.h>
+
+#include <CGAL/algorithm.h>
+#include <CGAL/boost/graph/iterator.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/Dynamic_property_map.h>
+#include <CGAL/property_map.h>
+#include <CGAL/Container_helper.h>
+
+#include <boost/range/value_type.hpp>
+#include <boost/range/reference.hpp>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+/// \ingroup PMP_repairing_grp
+///
+/// Adds the vertices and faces of a mesh into a (possibly non-empty) polygon soup.
+///
+/// \tparam PolygonMesh a model of `FaceListGraph`
+/// \tparam PointRange a model of the concepts `RandomAccessContainer` and
+///                    `BackInsertionSequence` whose value type can be constructed from
+///                    the point type of the polygon mesh
+/// \tparam PolygonRange a model of the concepts `RandomAccessContainer` and `BackInsertionSequence` whose
+///                      value type is itself a model of the concepts `RandomAccessContainer` and
+///                      `BackInsertionSequence` whose value type is `std::size_t`
+/// \tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+///
+/// \param mesh the mesh whose faces are being put in the polygon soup
+/// \param points points making the polygons of the soup
+/// \param polygons each element in the vector describes a polygon using the index of the points in `points`
+/// \param np optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+///
+/// \cgalNamedParamsBegin
+///   \cgalParamBegin{vertex_point_map}
+///     a model of `ReadablePropertyMap`, the property map with the points associated to the vertices of `mesh`.
+///     If this parameter is omitted, an internal property map for
+///     `CGAL::vertex_point_t` must be available in `PolygonMesh`.
```

Comment:
should it be available in `PolygonMesh`, or in `out`?
It seems to me that with dynamic property maps, that it is `out`